mobsf/MalwareAnalyzer/views/quark.py

```diff
@@ -24,38 +24,23 @@ def quark_analysis(app_dir, apk_file):
 
     from quark import config
     from quark.freshquark import download
-    from quark.Objects.quark import Quark
-    from quark.Objects.quarkrule import QuarkRule
+    from quark.report import Report
 
     logger.info('Running Quark %s', quark_ver)
     try:
+        json_report = ''
         # freshquark: update quark rules
         download()
 
         # default rules path
         rules_dir = Path(f'{config.HOME_DIR}quark-rules')
+        report = Report()
 
-        # Load APK
-        data = Quark(apk_file)
+        # Analyze apk
+        report.analysis(apk_file, rules_dir)
 
-        # Load rules
-        rules_list = rules_dir.rglob('*.json')
-
-        for rule_file in rules_list:
-            rule_checker = QuarkRule(rule_file.as_posix())
-
-            # Run the checker
-            data.run(rule_checker)
-
-            data.generate_json_report(rule_checker)
-
-        json_report = data.get_json_report()
-
-        # Clear all functools.lru_cache from quark
-        data.apkinfo.find_method.cache_clear()
-        data.apkinfo.upperfunc.cache_clear()
-        data.apkinfo.get_wrapper_smali.cache_clear()
-        data.apkinfo.construct_bytecode_instruction.cache_clear()
+        # Generate Report
+        json_report = report.get_report('json')
 
```

Comment:
The clear_cache() calls can explicitly release the resource.

```suggestion
        # Clear all functools.lru_cache from quark
        report.quark.apkinfo.find_method.cache_clear()
        report.quark.apkinfo.upperfunc.cache_clear()
        report.quark.apkinfo._construct_bytecode_instruction.cache_clear()
        report.quark.apkinfo.get_wrapper_smali.cache_clear()
```